File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.py
def check(num):
    tmp = 1
    while num>1:
        if num%2 == 1:
            tmp += 1
        num //= 2
    return tmp
def solution(n):
    tmp = check(n)
    for i in range(n+1,  1000001):
        if tmp == check(i):
            return i

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.java
import java.util.*;
class Solution {
    public int solution(int n) {
        int answer = 0;
        int target = check(n);
        for(int i=n+1; i<1000001; i++){
            if(check(i) == target){
                answer = i;
                break;
            }
        }
        return answer;
    }
    public static int check(int num){
        int count = 0;
        while(true){
            if(num == 1){
                count++;
                break;
            }
            if(num == 0){
                break;
            }
            if(num%2 == 1){
                count++;
            }
            num /= 2;
        }
        return count;
    }
}